src/GraphControl/Control/Grapher.h

```diff
@@ -155,6 +156,105 @@ public
         void PlotGraph();
         void AnalyzeEquation(GraphControl::Equation ^ equation);
 
+        // We can't use the EvalTrigUnitMode enum directly in as the property type because it comes from another module which doesn't expose
+        // it as a public enum class.  So the compiler doesn't recognize it as a valid type for the ABI boundary.
+        property int TrigUnitMode
+        {
+            void set(int value)
+            {
+                if (value != (int)m_solver->EvalOptions().GetTrigUnitMode())
+                {
+                    m_solver->EvalOptions().SetTrigUnitMode((EvalTrigUnitMode)value);
+                    PlotGraph();
+                }
+            }
+
+            int get()
+            {
+                return (int)m_solver->EvalOptions().GetTrigUnitMode();
+            }
+        }
+
+        property double XAxisMin
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().first;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(value, XAxisMax);
+                m_graph->GetOptions().SetDefaultXRange(newValue);
+                m_renderMain->RunRenderPass();
+            }
+        }
+
+        property double XAxisMax
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().second;
+            }
+            void set(double value)
+            {
+                std::pair<double, double> newValue(XAxisMin, value);
+                m_graph->GetOptions().SetDefaultXRange(newValue);
+                m_renderMain->RunRenderPass();
+            }
+        }
+
+        property double YAxisMin
+        {
+            double get()
+            {
+                return m_graph->GetOptions().GetDefaultXRange().first;
+            }
+            void set(double value)
```

Comment:
Is there a reason all of these properties do the business logic in the setter rather than in a callback method like the other properties?

Reply:
because these properties are normal, not dependency properties. The main reason is because they don't "host" the data, they are are a bridge to access to the data